### ISSUE TITLE
destroy many to many relationship with custom primaryKey in both tables

### ISSUE THREAD

**Issue by camel-chased** [balderdashy/waterline#783]
```javascript
//model user
schema        : true,
	autoPK        : false,
	autoCreatedAt : false,
	autoUpdatedAt : false,
attributes:{
    id:{
        primaryKey:true,
        columnName:'u_id'
    },
    houses:{
        collection:'house',
        via:'users'
    }
}

//model house
schema        : true,
	autoPK        : false,
	autoCreatedAt : false,
	autoUpdatedAt : false,
attributes:{
    id:{
        primaryKey:true,
        columnName:'h_id',
    },
    users:{
        collection:'user',
       via:'houses'
    }
}
```

try to destroy this with some houses/users in db (mysql)
ER_BAD_FIELD_ERROR: Unknown column 'undefined' in 'where clause'
when i change columnName to 'id' in both tables, everything goes nice

**Comment by esopian** [comment]
Same issue here.  Would love to find a solution for this.  Here's the mysql call that waterline tries to make.

```
SELECT `restaurantcuisine`.`ID`, `restaurantcuisine`.`RESTAURANT_ID`, `restaurantcuisine`.`CUISINE_ID` FROM `RESTAURANT_CUISINE` AS `restaurantcuisine`  WHERE `restaurantcuisine`.`RESTAURANT_ID` IN (undefined)  ORDER BY `restaurantcuisine`.`ID` DESC
```

**Comment by dmarcelino** [comment]
@camel-chased, can you try the fix in #928 to check if it fixes your issue please?

**Comment by camel-chased** [comment]
i don't work on waterline any more but i make something like to make it work: 
https://github.com/camel-chased/sails-redis/commit/53d7f4e584c9e40a210265b6e8c5d0368afe156d

**Comment by dmarcelino** [comment]
Thanks

**Comment by dmarcelino** [comment]
Fixed via #1022. A big thanks to @particlebanana! :wink:
